components/vc/rear/src/HW/HW_gpio_componentSpecific.c

```diff
@@ -11,44 +11,389 @@
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_BMS_LIGHT_EN] = {
+        .port = GPIOD,
+        .pin = GPIO_PIN_15,
+        .mode = GPIO_MODE_OUTPUT_PP,
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_IMD_LIGHT_EN] = {
         .port = GPIOC,
-        .pin = GPIO_PIN_13,
+        .pin = GPIO_PIN_6,
+        .mode = GPIO_MODE_OUTPUT_PP,
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_TSSI_G_EN] = {
+        .port = GPIOC,
+        .pin = GPIO_PIN_7,
+        .mode = GPIO_MODE_OUTPUT_PP,
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_TSSI_R_EN] = {
+        .port = GPIOC,
+        .pin = GPIO_PIN_8,
+        .mode = GPIO_MODE_OUTPUT_PP,
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_BR_LIGHT_EN] = {
+        .port = GPIOC,
+        .pin = GPIO_PIN_9,
         .mode = GPIO_MODE_OUTPUT_PP,
         .speed = GPIO_SPEED_FREQ_LOW,
         .pull = GPIO_NOPULL,
         .resetState = HW_GPIO_PINRESET,
     },
+    [HW_GPIO_CAN1_RX] = {
+        .port = GPIOA,
+        .pin = GPIO_PIN_11,
+        .mode = GPIO_MODE_INPUT,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_CAN1_TX] = {
+        .port = GPIOA,
+        .pin = GPIO_PIN_12,
+        .mode = GPIO_MODE_AF_PP,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_MCU_UART_EN] = {
+        .port = GPIOA,
+        .pin = GPIO_PIN_15,
+        .mode = GPIO_MODE_AF_PP,
+        .speed = GPIO_SPEED_FREQ_LOW,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_MCU_UART_TX] = {
+        .port = GPIOC,
+        .pin = GPIO_PIN_10,
+        .mode = GPIO_MODE_AF_PP,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_MCU_UART_RX] = {
+        .port = GPIOC,
+        .pin = GPIO_PIN_11,
+        .mode = GPIO_MODE_INPUT,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_SPI_CS_IMU] = {
+        .port = GPIOD,
+        .pin = GPIO_PIN_1,
+        .mode = GPIO_MODE_AF_PP,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,
+    },
+    [HW_GPIO_SPI_CS_SD] = {
+        .port = GPIOD,
+        .pin = GPIO_PIN_2,
+        .mode = GPIO_MODE_AF_PP,
+        .speed = GPIO_SPEED_FREQ_HIGH,
+        .pull = GPIO_NOPULL,
+        .resetState = HW_GPIO_PINRESET,

```

Comment:
Spi is typically active low chip select. Double check that you have these set properly. Also id have to double check the schematic but im pretty sure the cs lines are pulled up and should be open drain although i may be wrong